Missionframework/modules/00_init/scripts/server/markersUpdater/sectorMarkers.sqf

```diff
@@ -13,39 +13,55 @@
 
 waitUntil {KPLIB_save_loaded};
 
-// Create locked vehicle markers
+// Array to store sector and locked vehicle markers.
 private _lockedVehMarkers = [];
+// Shortcut to CfgVehicles.
 private _cfg = configFile >> "CfgVehicles";
 
+// Circle through all locked vehicle connections.
 {
+    // Classname of the locked vehicle.
     private _vehicle = _x select 0;
+    // Marker of the sector to which the vehicle is connected.
     private _base = _x select 1;
-    private _marker = createMarker ["lockedVehicle_" + _base, [markerPos _base select 0, (markerPos _base select 1) + 125]];
 
+    // Create marker above the sector marker.
+    private _marker = createMarker ["lockedVehicle_" + _base, [markerPos _base select 0, (markerPos _base select 1) + 125]];
     _marker setMarkerText (getText (_cfg >> _vehicle >> "displayName"));
     _marker setMarkerShape "ICON";
     _marker setMarkerType "mil_pickup";
     _marker setMarkerColor KPLIB_preset_colorEnemy;
 
+    // Storing of the created locked vehicle marker with relation to the base marker.
     _lockedVehMarkers pushBack [_marker, _base];
 } forEach KPLIB_sectors_lockedVeh;
 
-// Start loop to adjust marker colors
-private _sectorCount = -1;
+// Storage for the last blufor sectors count.
+private _bluforCount = -1;
+// Storage for the last active sectors count.
+private _activeCount = 0;
 
+// Loop to adjust marker colors during the game.
 while {KPLIB_campaignRunning} do {
```

Comment:
The "problem" is that marker colors are not JIP compatible. It could work as a "init colors" running on every client once and all the other changes after are supplied by the server commands, of course. But this needs steady publicVariables for each array, which is currently not necessary, as it's handled by the server completely. Indeed there is a need for some changes as soon as we go into "let HCs do some sector handling" etc.
Personally I would leave it this way and maybe keep it in mind for a possible "tweak" for the future dev process, if we could need some more tweaks then. Maybe this is also some kind of laziness at this evening, so I would really appreciate if you state a final opinion (with pointed to the JIP "issue" now) and I'll change it or leave it depending on your opinion :)

> What is not synchronized at JIP
> Markers (The markers themselves and their updated properties, e.g positions (if any)) 

Ref:
https://community.bistudio.com/wiki/6thSense.eu:EG#Join_in_Progress

Reply:
JIP "problem" clears why this loop is here. 

I think this is fine for now but in the future we could rewrite it. Keeping few public variables instead of using spawned script should make mission less resource intensive.

Reply:
The thing is that you're indeed right concerning "keep spawned scripts with endless loops as few as possible". I've also said it myself on the comment above. So I guess I won't put it to a later stage and will implement a "non loop" functionality for this tomorrow.
Thanks for the heads up and remind me on these things. It was getting late and I wanted to finish this task. Good you pointed that out to "keep the quality" of the refactoring. 👍 
And yeah, we could do this at a later stage but that would be a kind of refactor refactoring and we should do it right in the first attempt. ;)

Reply:
Maybe I were not right concerning JIP issue. Due to some more searching on this (I've wondered why the locked vehicle markers are synced for JIP then, as they're created by script by the server) I found this nice Dev posting in the forum. Maybe we even don't need a workaround for JIP (which is what I guess now).
Ref:
https://forums.bohemia.net/forums/topic/149271-created-markers-for-jip-players/?do=findComment&comment=2373385